### PR TITLE
Persistent FRP 

### PR DESCRIPTION
**Pull request by abidlabs**
Starting to work on a way to have persistent FRP connections. See [internal Slack discussion](https://huggingface.slack.com/archives/C02SPHC1KD1/p1675807052117969) for context

cc @XciD @Wauplin @aliabid94 

To use this version of Gradio for local testing:
* Check out this branch
* Run `cp gradio/templates/frontend/index.html gradio/templates/frontend/share.html`
* Create a Gradio interface with sharing enabled i.e.:

```py
import gradio as gr

gr.Interface(lambda x:x, "textbox", "textbox").launch(share=True)
```

It will hopefully fix the #3060 issue but I'd prefer to keep the issue open until the community has tested this fix

### CODE REVIEW

**Comment by abidlabs** [comment]
Released version `3.18.1b2` for easier testing. To test, do the following steps:

1. `pip install gradio==3.18.1b2`
2. Create a simple gradio demo like above with `share=True`
3. Leave it running for many hours and ensure that that the demo continues to run

File: gradio/tunneling.py [hunk 1]
```diff
@@ -85,7 +86,7 @@ def _start_tunnel(self, binary: str) -> str:
             f"{self.remote_host}:{self.remote_port}",
             "--disable_log_color",
         ]
-
+        print("Command:", " ".join(command))
```
Comment (freddyaboulton) [inline gradio/tunneling.py:89]: Do we want to keep this?

**Review by freddyaboulton — APPROVED** [review]
Tested on colab and my demo stayed up until colab disconnected me for inactivity (about 4 hours)! And the link generated in the colab is still valid.

Also verified that closing and launching and interface will preserve the same URL. 

Awesome!

**Comment by abidlabs** [comment]
> And the link generated in the colab is still valid.

Yeah same for me... dunno how :D

File: gradio/tunneling.py (hunk 1 above)
Comment (abidlabs) [reply gradio/tunneling.py:89]: Will remove, thanks!

**Comment by abidlabs** [comment]
Has been working well for me too, so I'll go ahead and merge. Let's release a new version soon and have the community test this out. Thanks @Wauplin and @XciD for your help on this PR!

**Pull request merged**